jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/JMSCore.java

```diff
@@ -0,0 +1,344 @@
+        }
+        
+        // Make sure JMS is safe to use (has not been closed)
+        if (localJMS == null) {
+            log.error("JMS connection closed before operation could complete");
+        } else {
+            Object msg;
+            String dest;
+            String msgFormat;
+            boolean isQueue;
+            
+            // Getting the contents of the message if specified, or defaulting to an empty string
+            if (request.get("message") instanceof String) {
+                msg = (String) request.get("message");
+            } else if (request.get("message") instanceof Map) {
+                msg = (Map) request.get("message");
+            } else {
+                msg = null;
+                log.debug("No message was specified in the publish request, or the message was not a String/Map. "
+                        + "The message was set to its default value, null.");
+            }
+                
+            // Getting the destination of the message
+            if (request.get("queue") instanceof String) {
+                dest = (String) request.get("queue");
+                isQueue = true;
+            } else if (request.get("topic") instanceof String) {
+                dest = (String) request.get("topic");
+                isQueue = false;
+            } else {
+                log.error("No destination was specified, or destination was not a String. Either a topic "
+                        + "or a queue must be included as a String in the publish request.");
+                return;
+            }
+            
+            // Getting the message format if it was specified, or defaulting to Message
+            if (request.get("JMSFormat") instanceof String) {
+                msgFormat = (String) request.get("JMSFormat");
+            } else {
+                log.debug("No JMSFormat was specified, the default Message message type will be used.");
+                msgFormat = "Message";
+            }
+            
+            // Sending the message to the appropriate destination
+            try {
+                localJMS.produceMessage(msg, dest, msgFormat, isQueue);
+            } catch (JMSException e) {
+                log.error("An error occured when attempting to send the given message.", e);
+            } catch (DestinationNotConfiguredException e) {
+                log.error("An error occured when attempting to send the given message. The source was not configured "
+                        + "to send messages to the following destination: " + dest + ".", e);
+            } catch(UnsupportedJMSMessageTypeException e) {
+                log.error("An error occured when attempting to send the given message. The provided JMS Message Type: "
+                        + e.getMessage() + " is either invalid or not currently supported.");
+            } catch (Exception e) {
+                log.error("An unexpected error occured when attempting to send the given message.", e);
```

Comment:
We throw a DestinationNotConfiguredException if we could not find a message producer/consumer for a given queue or topic. In the case that Roger was referring to earlier, I believe this would throw a null pointer exception which would be caught in the last catch block. 
What we could do there is check that ```localJMS == null```, and log/send a more specific error message in that case. Would that work?